packages/core/src/agent/message-list.ts

```diff
@@ -0,0 +1,571 @@
+import { randomUUID } from 'crypto';
+import type { CoreMessage, FilePart, Message, UIMessage } from 'ai';
+import type { MessageType } from '../memory';
+import { isCoreMessage, isUiMessage } from '../utils';
+
+export type MastraMessageContentV2 = {
+  format: 2; // format 2 === UIMessage in AI SDK v4
+  // TODO: When we bump to AI SDK v5 and make "format: 3" we might need to inline these types with a copy/paste
+  parts: UIMessage['parts'];
+  experimental_attachments?: UIMessage['experimental_attachments'];
+  content?: UIMessage['content'];
```

Comment:
This will make it message.content.content, right?

Reply:
yeah it will unfortunately! In AI SDK v5 there is no content field though, so it'll go away soon